crates/bevy_window/src/system.rs

```diff
@@ -24,3 +25,30 @@ pub fn exit_on_window_close_system(
         app_exit_events.send(AppExit);
     }
 }
+
+pub fn update_cursor_position(
```

Comment:
Random thought: we could handle these events in bevy_winit

pros: window state is correct _before_ schedule execution starts (which means systems in PRE_UPDATE can use the cursor position). one less system running

cons: more backend-specific logic

I personally think moving this into bevy_winit's window event handler is the right call, but I'm curious what y'all think.

Reply:
Sounds like a good idea, I'm excited to have bevy get rid of the hacky way of doing cursor pos currently.

Reply:
Do you mean that the code residing in `bevy_winit` would modify `bevy_window`'s resources?

Reply:
Yup! We already do that in a few places

Reply:
What about other windowing plugins? SDL, miniquad?
Should they also touch `bevy_window` internals instead of sending messages and letting `bevy_window` handle these as it fits?

Reply:
I totally see your point and for most things I would just agree outright (because the design you're suggesting is "cleaner"). I still think we should make an exception here for a few reasons:

1. The design as it stands is already "unclean" in some respects. The winit backend already mutates bevy_window state for a number of scenarios: creating windows, resizing windows, touch events, scale factor changes
2. Having access to up to date window state as early as possible seems valuable 
3. New window backends are relatively uncommon (and are generally built by proficient people), so I'm fine eating extra complexity of moderate size there to make user facing scenarios nicer.

Reply:
And I do see your point. 
So, while I'm working with this code already, maybe I should take an opportunity and change other messaging like this and move it to direct mutation? Where it makes sense.

Reply:
I'm sold!

Reply:
I added `update_cursor_position_from_backend` like other `update_*_from_backend` methods and removed the system reading messages.

I have a feeling that mouse handling in my game works smoother now. :smile: